### PR TITLE
Add SARI metric

### PR DESCRIPTION
**Pull request by ddhruvkr**
Made changes to add the SARI evaluation metric for text simplification. Haven't made changes for SARI for the case of the submission file (submission.json) since I was not sure if it was finalized yet.

### CODE REVIEW

**Comment by tuetschek** [comment]
@ddhruvkr : This looks good to me – thanks! Are there any changes needed in the submission.json file? I guess we'll just need source files to comparison?

**Comment by ddhruvkr** [comment]
@tuetschek Yeah, that should be the only change to the submission.json file. And I guess also the corresponding changes in the run_metric.py and gem_metrics/__init__.py file right?

**Comment by tuetschek** [comment]
@ddhruvkr : In that case, we don't need any changes to the example submission.json file – that only contains the outputs. But yeah, we will need changes to `run_metrics.py` and `gem_metrics/__init__.py`.

Actually, we could load the sources and the references from the same files – if we have the “source” and “target” keys in the same file, it should work. (Any opinions on the name of that entry? I've already used “input” in a few places... but it's easy to change.)

**Comment by tuetschek** [comment]
@ddhruvkr would you be able to provide me with sources & refs for the simplification datasets? I'll upload them here: https://github.com/GEM-benchmark/GEM-metrics/releases/tag/data 

The Turk set is there already, but without sources.

**Comment by ddhruvkr** [comment]
@tuetschek Sure, we could just keep one file for loading the sources and references. In the new file for loading source sentences, I used "source" as the name in the JSON object. I am fine with using either "source" or "input". 

Regarding the datasets, we might normalize the sentences in all three simplification datasets. Once we have finalized this (should be a couple of days) I'll share it here.

**Comment by tuetschek** [comment]
@ddhruvkr : Great – I can rename mine to "source", it's not a problem :-). Let me know when you have the data 👍

**Comment by ddhruvkr** [comment]
@tuetschek Sorry for the late response.

We plan to run two setups for simplification: 
(1) Simplification focused on all 3 operations (paraphrasing, splitting, deletion). 
Training data is Wiki-Auto with "auto_full_with split" config.
Reference Sentences: https://github.com/chaojiang06/wiki-auto/raw/master/wiki-auto/GEM2021/full_with_split/train.dst
Source Sentences: https://github.com/chaojiang06/wiki-auto/raw/master/wiki-auto/GEM2021/full_with_split/train.src

Dev and Test data for this setup is Asset dataset.

(2) Simplification focused on only paraphrasing
Train data is Wiki-Auto with "auto_full_no split" config.
Reference Sentences: https://github.com/chaojiang06/wiki-auto/raw/master/wiki-auto/GEM2021/full_no_split/train.dst
Source Sentences: https://github.com/chaojiang06/wiki-auto/raw/master/wiki-auto/GEM2021/full_no_split/train.src

Dev and Test data for this setup is Turk dataset.

The Source sentences for Turk and Asset datasets are the same.

I saw that we already have json files for the reference sentences of Turk dataset. Could you create them again with the files from the following link (https://github.com/cocoxu/simplification/tree/master/data/turkcorpus/GEM)?
This location also the validation and test src sentences for Turk (and Asset) (.norm files).

Please let me know if I missed something or if it's unclear.

**Comment by tuetschek** [comment]
@ddhruvkr no worries! I can do that, but I'm not really sure how:
- I assume “tune.*” means validation and “test.*” is testing?
- What's the difference between “*.norm” and “*.turk.[0-7]”?
- Are these just references, or are there sources somehow, too?
- I'm assuming I don't need to do anything with the train sets, right?

**Comment by ddhruvkr** [comment]
@tuetschek 
* Yeah, tune.* means validation and test.* is testing.
* *.norm are the source sentences (these are the source sentences for both Asset and Turk) and *.turk.[0-7] are the reference sentences for the Turk dataset. Both Asset and Turk have multiple reference sentences for each source sentence (For Turk, each source sentence has 8 reference sentences, hence 0-7). So Turk and Asset take the same source sentences and then collect reference sentences differently (Turk has only paraphrasing while Asset uses splitting and deletion as well). Thus, we get different references for the same source sentences with Asset and Turk. We do not need to make any changes to the reference validation and test sentences for Asset for which json files are already created. For Turk, we made some changes related to the normalization of sentences and therefore the json files for the reference sentences can be recreated.
* For the training set, we use WikiAuto. Here, each source sentence has a single reference sentence. We create two versions for Wikiauto to be used as training sets. The models trained using the first version ("auto_full_with split" config) will be evaluated on Asset and the models trained on the second split ("auto_full_no split" config) will be evaluated on Turk. For Wikiauto, .src are the source sentences and .dst are the simplified sentences.

Sorry for the confusion. Please let me know if you have any more questions.

**Comment by tuetschek** [comment]
@ddhruvkr check out https://github.com/GEM-benchmark/GEM-metrics/releases/download/data/turk_test.json and https://github.com/GEM-benchmark/GEM-metrics/releases/download/data/turk_val.json . 
Also from 5dd786d3dd4729ab6fd306eaf6069e8a04540d1f SARI should now be computed for Turk by default.

**Comment by ddhruvkr** [comment]
@tuetschek This looks great :) Thanks!
Since you added the source sentences to the original files itself, the asset validation and test json files can be modified as well to add the source sentences.

**Comment by tuetschek** [comment]
@ddhruvkr Done – hope it's OK like so:
https://github.com/GEM-benchmark/GEM-metrics/releases/download/data/asset_test.json 
https://github.com/GEM-benchmark/GEM-metrics/releases/download/data/asset_val.json

**Comment by ddhruvkr** [comment]
@tuetschek Thanks. This looks great!
